Dockerfile

```
@@ -0,0 +1,30 @@
+#Builder Stage
+FROM gradle:7.6.0-jdk17 AS builder
+WORKDIR /app
+
+COPY build.gradle settings.gradle ./
+COPY gradle ./gradle
+COPY gradlew ./
+RUN chmod +x ./gradlew
+
+RUN ./gradlew dependencies --no-daemon
+
+COPY src ./src
+RUN ./gradlew -x test build --no-daemon
+
+#Runtime Stage
+FROM amazoncorretto:17-alpine
+WORKDIR /app
+
+ENV SPRING_PROFILES_ACTIVE=prod
+ENV PROJECT_NAME=discodeit
+ENV PROJECT_VERSION=1.2-M8
+ENV JVM_OPTS=""
+
+COPY --from=builder /app/build/libs/${PROJECT_NAME}-${PROJECT_VERSION}.jar app.jar
+
+
+EXPOSE 80
+
+ENTRYPOINT ["sh", "-c"]
+CMD ["java $JVM_OPTS -jar app.jar"]
```

```suggestion
ENTRYPOINT ["sh", "-c", "exec", "java $JVM_OPTS -jar app.jar"]
```
성근님 방식대로 한다면 1번 프로세스는 쉘이 될겁니다. 아래 내용 참고해보세요.
https://brunch.co.kr/@growthminder/142

shell form vs exec form 에 대해 알고 계셨으면 좋겠습니다.
그리고 왜 exec form을 써야 하는지도 알고 계셨으면 좋겠어요!
https://engineering.pipefy.com/2021/07/30/1-docker-bits-shell-vs-exec/